### PR TITLE
For SG-13320: shared_thumbnail  raise a specific error

### PR DESCRIPTION
**Pull request by daigles**
* When called for a source entity for which the thumbnail is not ready

### CODE REVIEW

File: shotgun_api3/shotgun.py [hunk 1]
```diff
@@ -2250,14 +2256,16 @@ def share_thumbnail(self, entities, thumbnail_path=None, source_entity=None,
 
         result = self._send_form(url, params)
 
-        if not result.startswith("1"):
-            raise ShotgunError("Unable to share thumbnail: %s" % result)
-        else:
+        if result.startswith("1:"):
```
Comment (willis102) [inline shotgun_api3/shotgun.py]: I notice this changed from testing `startswith("1")` to `startswith("1:")`, but below we're just looking for `startswith("2")` -- should these two tests be consistent?

File: shotgun_api3/shotgun.py [hunk 2]
```diff
@@ -136,6 +136,12 @@ class ShotgunFileDownloadError(ShotgunError):
     """
     pass
 
+class ShotgunThumbnailNotReady(ShotgunError):
+    """
+    Exception for file download-related errors.
```
Comment (willis102) [inline shotgun_api3/shotgun.py]: Do we want to be more specific about what this Exception is for (maybe something like "attempting to download or use a pending thumbnail") or do you see it as being used more generally as well down the road?

File: shotgun_api3/shotgun.py (hunk 1 above)
Comment (willis102) [reply shotgun_api3/shotgun.py]: Disregard this -- I just read your PR on the server-side code and see why this is.

File: README.md [hunk 3]
```diff
@@ -100,8 +100,12 @@ Additionally, when testing locally, tests should be run for both python 2 and py
 
 Integration and unit tests are provided.
 
-- All tests require the [nose unit testing tools](http://nose.readthedocs.org), and a `tests/config` file (you can copy an example from `tests/example_config`).
-- Tests can be run individually like this: `nosetest tests/test_client.py`
+- All tests require:
+    - The [nose unit testing tools](http://nose.readthedocs.org),
+    - Running `pip install -r tests/ci_requirements.txt`
```
[the remark this answers is not in the document]
Comment (daigles) [reply README.md]: done!

File: README.md [hunk 4]
```diff
@@ -100,8 +100,12 @@ Additionally, when testing locally, tests should be run for both python 2 and py
...
+    - A `tests/config` file (you can copy an example from `tests/example_config`).
+- Tests can be run individually like this: `nosetest --config="nose.cfg" tests/test_client.py`
+    - Make sure to not forget the `--config="nose.cfg"` option
```
[the remark this answers is not in the document]
Comment (daigles) [reply README.md]: done!

File: shotgun_api3/__init__.py [hunk 5]
```diff
@@ -8,7 +8,8 @@
 # agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
-from .shotgun import (Shotgun, ShotgunError, ShotgunFileDownloadError, Fault,  # noqa unused imports
+from .shotgun import (Shotgun, ShotgunError, ShotgunFileDownloadError,
```
[the remark this answers is not in the document]
Comment (daigles) [reply shotgun_api3/__init__.py]: thanks! I was wondering about that!

File: shotgun_api3/shotgun.py (hunk 2 above)
Comment (daigles) [reply shotgun_api3/shotgun.py]: Good catch! I totally forgot to update the description. Done!

**Pull request merged**